### PR TITLE
Feature/timer

### PR DESCRIPTION
**Pull request by philalewis**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (change which fixes an issue)
- [x] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

 
Build timer component
 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# Where should reviewers start?

# How Has This Been Tested?
no
Please check the option(s) that best describe your tests. 

- [ ] New spec files were added
- [ ] Tests were added to existing spec files
- [ ] Some tests were changed
- [x] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Please add further details of your tests here.

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have removed all unused code or commented why I left it
- [ ] I have deployed my changes to a test-heroku and they work
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by Jayquintana — APPROVED** [review]
looks great, need to merge this before I can finish working on the WritingPage branch !

File: src/Components/Timer/Timer.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  useEffect(() => {
+    if (elapsedTime < total_seconds - 30) {
+      colorClass = 'green'
+    } else if (elapsedTime >= total_seconds - 30 && elapsedTime < total_seconds) {
+      colorClass = 'yellow'
+    } else if (elapsedTime >= total_seconds) {
+      colorClass = 'red'
+    }
+  }, [elapsedTime])
```
Comment (gjtrecny) [inline src/Components/Timer/Timer.tsx:35]: Nice work on useEffect!

Comment (gjtrecny) [inline src/Components/Timer/Timer.tsx:35]: Nice work on useEffect!

File: src/Components/Timer/Timer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const toggleTimer = () => {
+    toggleInProgress(!inProgress)
+  }
+
+  const formatTime = () => {
+    const minutes = Math.floor(elapsedTime / 60)
```
Comment (gjtrecny) [inline src/Components/Timer/Timer.tsx:42]: Not exactly sure why you're using Math.floor here. Could you explain in the comments?

Comment (gjtrecny) [inline src/Components/Timer/Timer.tsx:42]: Not exactly sure why you're using Math.floor here. Could you explain in the comments?

**Review by gjtrecny — APPROVED** [review]

File: src/Components/Timer/Timer.tsx (hunk 2 above)
Comment (philalewis) [reply src/Components/Timer/Timer.tsx:42]: For the minutes, we want to divide the number of seconds by 60 and then round down. Seconds are calculated seperately, so we don't want minutes to be decimal.

File: src/Components/Timer/Timer.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+interface Props {
+  total_seconds: number,
+  stopTimer: (arg0: string) => {}
+}
+
+const Timer: React.FC<Props> = ({ total_seconds, stopTimer }) => {
+  const [ elapsedTime, setElapsedTime ] = useState(0)
+  const [ inProgress, toggleInProgress ] = useState(false)
+  let colorClass: string = 'green'
```
Comment (Alynn022) [inline src/Components/Timer/Timer.tsx:12]: When you're assigning the colorClass here, is this being used for CSS too? Really curious as to how the colorClass works!

File: src/Components/Timer/Timer.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  const formatTime = () => {
+    const minutes = Math.floor(elapsedTime / 60)
+    const seconds = elapsedTime % 60
+    return `${minutes}:${seconds}`
+  }
+
+  return (
+    <section className={colorClass}>
```
Comment (Alynn022) [inline src/Components/Timer/Timer.tsx:48]: I see that we're assigning the className of the color here!

File: src/Components/Timer/Timer.tsx (hunk 3 above)
Comment (philalewis) [reply src/Components/Timer/Timer.tsx:12]: Yes, it will be used in the CSS, though I have not written any of that yet.

Comment (Alynn022) [inline src/Components/Timer/Timer.tsx:12]: Did typescript error out when you didn't type the variable colorClass (colorClass: string) ? I'm just curious!

**Review by Alynn022 — APPROVED** [review]

File: src/Components/Timer/Timer.tsx (hunk 3 above)
Comment (philalewis) [reply src/Components/Timer/Timer.tsx:12]: No, I don't believe so. I assumed I needed to type it from the start. Would it be better in this case to just use the implied type?

**Pull request merged**